__test__/shop.test.js

```diff
@@ -0,0 +1,89 @@
+var Shop = require('../src/shop');
+var Item = require('../src/item');
+
+describe("Gilded Rose", function() {
+  it('Name does not change after update', () => {
+    const gildedRose = new Shop([ new Item("foo", 0, 0) ]);
+    const items = gildedRose.updateQuality();
+    expect(items[0].name).toBe("foo");
+  });
+
+  it('Quality degrades twice as fast after sellIn is zero', () => {
+    const gildedRose = new Shop([ new Item("foo", -1, 5) ]);
+    const items = gildedRose.updateQuality();
+    expect(items[0].quality).toBe(3);
+    expect(items[0].sellIn).toBe(-2);
+  });
+
+  it('Quality should never be negative', () => {
+    const gildedRose = new Shop([ new Item("foo", 0, 0), new Item("foo2", -1, 1) ]);
+    const items = gildedRose.updateQuality();
+    expect(items[0].quality).toBe(0);
+    expect(items[1].quality).toBe(0);
+  });
+
+  it('Aged Brie should always increase in value', () => {
+    const gildedRose = new Shop([ new Item("Aged Brie", 0, 0) ]);
+    const prevQuality = gildedRose.items[0].quality; 
+    const items = gildedRose.updateQuality();
+    expect(items[0].sellIn).toBe(-1);
+    expect(items[0].quality).toBeGreaterThan(prevQuality);
+  });
+
+  it('Aged Brie should increase by 1', () => {
+    const gildedRose = new Shop([ new Item("Aged Brie", 3, 1) ]);
+    const prevQuality = gildedRose.items[0].quality; 
+    const items = gildedRose.updateQuality();
+    expect(items[0].sellIn).toBe(2);
+    expect(items[0].quality).toBe(2);
+  });
+
+  it('Quality never goes above 50', () => {
+    const gildedRose = new Shop([ new Item("Aged Brie", 0, 50) ]);
+    const items = gildedRose.updateQuality();
+    expect(items[0].quality).toEqual(50);
+  });
+
+  it('Sulfuras never change their sellIn or Quality', () => {
```

Comment:
It is common practice to have one assertion per test. Ideally, we want the unit test to fail for one reason and one reason only.

Reply:
Okay I split it into 2 separate tests.